presto-teradata-functions/src/main/java/com/facebook/presto/teradata/functions/dateformat/DateFormatParser.java

```diff
@@ -70,9 +102,22 @@ public static DateTimeFormatter createDateTimeFormatter(String format)
                             String.format("Failed to tokenize string [%s] at offset [%d]", token.getText(), token.getCharPositionInLine()));
             }
         }
-
         try {
-            return builder.toFormatter();
+            // Append default values(0) for time fields(HH24, HH, MI, SS) because JSR-310 does not accept bare Date value as DateTime
+
+            if (formatContainsHourOfAMPM) {
+                // At the moment format does not allow to include AM/PM token, thus it was never possible to specify PM hours using 'HH' token in format
+                // Keep existing behaviour by defaulting to 0(AM) for AMPM_OF_DAY if format string contains 'HH'
+                builder.parseDefaulting(ChronoField.HOUR_OF_AMPM, 0)
```

Comment:
Could we set all three of the parseDefaultings regardless of whether the format contains hour of AMPM (if a format doesn't have HOUR_OF_AMPM, then this won't be relevant, and similarly for HOUR_OF_DAY if it does have HOUR_OF_AMPM)?

Reply:
Unfortunately, not. 
Conflicts can happen if both HOUR_OF_DAY and HOUR_OF_AMPM values provided (e.g. one with parseDefaultings and another in the input).
Example: to_timestamp('1988/04/08 2','yyyy/mm/dd hh') will trigger "Conflict found: HourOfDay 0 differs from HourOfDay 2" error if we set all three of the parseDefaultings